### PR TITLE
[LOOP-4464] only round bolus volumes down

### PR DESCRIPTION
**Pull request by nhamming**
https://tidepool.atlassian.net/browse/LOOP-4464

### CODE REVIEW

File: Loop/Managers/DeviceDataManager.swift [hunk 1]
```diff
@@ -1276,7 +1276,8 @@ extension DeviceDataManager: LoopDataManagerDelegate {
             return units
         }
 
-        let rounded = ([0.0] + pumpManager.supportedBolusVolumes).enumerated().min( by: { abs($0.1 - units) < abs($1.1 - units) } )!.1
+        // only allow volumes that are rounded down
+        let rounded = ([0.0] + pumpManager.supportedBolusVolumes).enumerated().min( by: { abs($0.1 - units) < abs($1.1 - units) && $0.1 < units } )!.1
```
Comment (Camji55) [inline Loop/Managers/DeviceDataManager.swift]: Is the force unwrap safe here?

Comment (Camji55) [reply Loop/Managers/DeviceDataManager.swift]: Also, is enumeration needed here? It doesn't seem like the index is being used at all. Would this work? 👇🏻 

```suggestion
        let rounded = ([0.0] + pumpManager.supportedBolusVolumes).min(by: { abs($0 - units) < abs($1 - units) && $1 < units })!
```

Comment (ps2) [inline Loop/Managers/DeviceDataManager.swift]: PumpManagers already expose a rounding method `roundToSupportedBolusVolume`.  I think we should probably be using that.

**Review by ps2 — CHANGES_REQUESTED** [review]

File: Loop/Managers/DeviceDataManager.swift (hunk 1 above)
Comment (nhamming) [reply Loop/Managers/DeviceDataManager.swift]: yes, I also thought this code was unusual.

Comment (nhamming) [reply Loop/Managers/DeviceDataManager.swift]: Thanks. Forgot about that. Updating

**Review by ps2 — APPROVED** [review]
LGTM!

**Pull request merged**